### PR TITLE
mysql/decimal: fix bug in do_div_mod

### PR DESCRIPTION
**Pull request by bb7133**
## What have you changed? (mandatory)

Fixed a bug in decimal modulo operation(do_div_mod in mysql/decimal.rs)

## What are the type of the changes? (mandatory)

- Bug fix

## How has this PR been tested? (mandatory)

Added some UT cases

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

Corresponding bug-fix at TiDB side: https://github.com/pingcap/tidb/pull/7113

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_arithmetic.rs [hunk 1]
```diff
@@ -792,9 +792,9 @@ mod test {
             ),
             (
                 ScalarFuncSig::ModDecimal,
-                str2dec("-33"),
-                str2dec("-11"),
-                str2dec("0"),
+                str2dec("0.0000000001"),
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_arithmetic.rs]: Could we keep the old test?

**Comment by AndreMouche** [comment]
@breeswish PTAL

File: src/coprocessor/dag/expr/builtin_arithmetic.rs (hunk 1 above)
Comment (bb7133) [reply src/coprocessor/dag/expr/builtin_arithmetic.rs]: Sorry, this is a mistake @AndreMouche

**Comment by coocood** [comment]
LGTM

File: src/coprocessor/codec/mysql/decimal.rs [hunk 2]
```diff
@@ -688,7 +688,15 @@ fn do_div_mod(
             buf[l_idx] = dcarry as u32;
         }
         idx_to = 0;
-        let mut int_word_to = word_cnt!(l_prec.wrapping_sub(l_frac_cnt) as i8, i8) - l_idx as i8;
+
+        int_cnt_to = l_prec.wrapping_sub(l_frac_cnt) as i8 - l_idx as i8 * DIGITS_PER_WORD as i8;
+
+        let mut int_word_to = if int_cnt_to < 0 {
+            int_cnt_to / DIGITS_PER_WORD as i8
```
Comment (breeswish) [inline src/coprocessor/codec/mysql/decimal.rs]: shall we integrate this logic to word_cnt! instead? @AndreMouche

Comment (AndreMouche) [reply src/coprocessor/codec/mysql/decimal.rs]: I think we'd better keep compatible with `TIDB` here @breeswish

**Comment by AndreMouche** [comment]
@breeswish PTAL

**Comment by AndreMouche** [comment]
Please `make format` and fix the `DCO`

**Comment by AndreMouche** [comment]
Please fix the errors in CI

**Comment by bb7133** [comment]
Fixed, PTAL @AndreMouche

**Pull request merged**